core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountActor.scala

```diff
@@ -111,15 +111,19 @@ final case class WorkspaceBillingAccountActor(dataSource: SlickDataSource, gcsDA
   private def syncBillingProjectWithGoogle[F[_]](billingProject: RawlsBillingProject)
                                                 (implicit R: Ask[F, BillingAccountChange], M: Monad[F], L: LiftIO[F])
   : F[Unit] =
-    for {
-      // only v1 billing projects are backed by google projects
-      isV1BillingProject <- L.liftIO {
-        gcsDAO.rawlsCreatedGoogleProjectExists(billingProject.googleProjectId).io
```

Comment:
Almost!!! This only happens for v2 billing projects because their name doesn't have to be a valid google project id. When we ask google about a billing project with an invalid name we get a 400 back and then this actor marks the attaempt to sync the billing project as failed. Note that billing account changes cannot be applied to v1 billing projects that have not been completed successfully.

`billingProject.googleProjectId` *always* returns the billing project name wrapped in a `GoogleProjectId` (really this should be `Option[GoogleProjectId]` but unfortunately we have no good way of telling that an arbitrary billing project is v1 or v2).

Does that help?

Reply:
I see, so we're storing the name of the billing project as the `googleProjectId` even for v2 billing projects.

I think we could still run into this issue even with this fix, if a v2 billing project has a name that is a valid v1 billing project name?

Could we check for / handle the (400) error we expect?

Reply:
oh, I didn't realize (or forgot) that's what `googleProjectId` did.

what are your thoughts on this:

> I think we could still run into this issue even with this fix, if a v2 billing project has a name that is a valid v1 billing project name?

I can see this change being worth merging because it probably covers most of the projects. 

I'm not sure if this will make it into the monolith release though, since it is scheduled to go out at 4pm.

Alternatively, we could revert the change, but ideally we won't need to..

Reply:
If the billing project name is a valid google project id then google will give us a 404 or 403 when we try to access a google project with the same name.

Reply:
If we wanted to make doubly-sure that it's a google project attributed to a billing project then we need to ask sam. I don't think we can't do that yet as we don't have the admin apis.

Reply:
I've just realised that this is a potential security vulnerability. We should back out the original change.